modules/gltf/gltf_document.cpp

```diff
@@ -3258,13 +3316,26 @@ GLTFTextureIndex GLTFDocument::_set_texture(Ref<GLTFState> state, Ref<Texture2D>
 	return gltf_texture_i;
 }
 
-Ref<Texture2D> GLTFDocument::_get_texture(Ref<GLTFState> state, const GLTFTextureIndex p_texture) {
+Ref<Texture2D> GLTFDocument::_get_texture(Ref<GLTFState> state, const GLTFTextureIndex p_texture, bool p_is_normal) {
 	ERR_FAIL_INDEX_V(p_texture, state->textures.size(), Ref<Texture2D>());
 	const GLTFImageIndex image = state->textures[p_texture]->get_src_image();
 
 	ERR_FAIL_INDEX_V(image, state->images.size(), Ref<Texture2D>());
-
-	return state->images[image];
+	if (state->handle_binary_image != GLTFState::GLTF_DOCUMENT_EMBED_AS_BASISU) {
+		return state->images[image];
+	}
+	if (!p_is_normal) {
+		return state->images[image];
+	}
```

Comment:
This should be cached for both normal and non-normal cases.

If a normal-map is used 100 times or even twice, we cannot afford to make duplicate copies of the texture in this case.

Reply:
I think a full caching system is material for a different pr, need to make sure it applies to everything and this pr is already dense.

Reply:
I misread the comment. It now caches.